### DISCUSSION TITLE
Dockerfile to cross compile for linux+mac+windows

### DISCUSSION THREAD

**Discussion by santrancisco** [AllenDang/giu#201] [Show and tell]
I hope this may help someone and save them sometime when starting to play with giu. Today i wanted to play with giu, compiling it in my host (Linux) took sometimes but i got there in the end, then i wanted to try cross-compile for windows... then MacOSX... long story short, i ended up with a bunch of dependencies installation, trials and errors, testing with various compiling flags, env var and after successfully compiling them all in my host machine, I went through my bash history and started documenting everything into a Dockerfile what i needed and what worked for me.

What came out at the end is a Docker container that let me cross compile between Mac, Linux and Windows with the examples found here. [Here is the gist](https://gist.github.com/santrancisco/6fd1e84b57a43ae1eb675614b02bacac) where you can find my notes with Dockerfile to build a docker container with all the toolchains, a Makefile with all the flags/switches i used to compile for Mac, Nix and Windows app. I learnt a few tricks along the way like "-Hwindowsgui" makes the command prompt in windows goes away and adding ".app" extension do the same for MacOSX app.

Anyhow, Thanks @AllenDang  for all the hard work! I'm looking forward to trial this in some small projects at work.

**Comment by AllenDang** [comment]
@santrancisco Big thanks!!!! I always want to create dockerfiles to cross-compile, but I'm not familiar with docker. Do you think we could create some ready to use standard dockerfiles, and update the README, so everyone to could easily pull them anytime?

**Comment by gucio321** [comment]
Hi @AllenDang @santrancisco 
I think, I did some progress here. I am able to run my giu app inside docker container. It means, that I can easly build it. Cross-compilation shouldn't be a big deal now. 

here is a result:
![image](https://user-images.githubusercontent.com/73652197/201974727-00c2e0e6-fb16-40df-93b9-5b0e9e219245.png)

**NOTE** I'm using podman since it is compatible with fedora, but docker should work 

[here](https://github.com/gucio321/tic-tac-go) is my project using docker
